src/createInspector.ts

```diff
@@ -203,6 +205,32 @@ export function createInspector<TAdapter extends Adapter>(
   return inspector;
 }
 
+function deepCopyAndPrepareForSafeStringify(
+  obj: any,
+  hash = new WeakMap()
+): any {
+  if (typeof obj === 'bigint') return obj.toString();
```

Comment:
I think this is pretty ad-hoc. Ideally, the serialization should try to avoid losing information. It is a different thing to have a `"12"` string and a `12n` bigint value. The serialization should try to encode the encoded's type using a wrapper object or something so its decoder counterpart could try to deserialize it closely. Or at least display as much as possible about the original value. 

Some inspiration here could be drawn from libraries like `superjson`, React Flight's serialization algorithm and similar.

Reply:
Note that the lossy nature of the current approach has already been raised as a concern here: https://github.com/statelyai/inspect/pull/46#issuecomment-3508658355